src/files/glob-source.js

```diff
@@ -0,0 +1,89 @@
+'use strict'
+
+const fs = require('fs-extra')
+const glob = require('it-glob')
+const Path = require('path')
+const errCode = require('err-code')
+
+/**
+* Create an async iterator that yields paths that match requested file paths.
+*
+* @param {String} ...paths File system path(s) to glob from
+* @param {Object} [options] Optional options
+* @param {Boolean} [options.recursive] Recursively glob all paths in directories
+* @param {Boolean} [options.hidden] Include .dot files in matched paths
+* @param {Array<String>} [options.ignore] Glob paths to ignore
+* @param {Boolean} [options.followSymlinks] follow symlinks
+* @yields {Object} File objects in the form `{ path: String, content: AsyncIterator<Buffer> }`
+*/
+module.exports = async function * globSource (...args) {
```

Comment:
can't the arguments be `function(Array<string> paths, Object options)` ? instead of multiple paths as arguments?

Reply:
It's just been ported as-is from `js-ipfs`